### PR TITLE
Reconfigure test environment

### PR DESCRIPTION
**Pull request by piotrekpasciak**
This PR consists of:
- moving Capybara, FactoryBot and Warden configs to separate files
- replacing DatabaseCleaner with fixtures
- removal of rspec-retry gem
- removal of capybara-email gem
- fixing of wrong domain in URL issue
- tidy up of spec helpers
- tidy up of rails_helper.rb file
- adding vcr gem and config
- using VCR for requests done by NotebookService in test environment

### CODE REVIEW

File: spec/services/notebook_service_spec.rb [hunk 1]
```diff
@@ -1,11 +1,17 @@
 require 'rails_helper'
 
 RSpec.describe NotebookService do
-
   let(:notebook_url) { "https://gitlab.crowdai.org/crowdai-dojo/Selecting-data-in-pandas/raw/master/index.ipynb"}
 
   describe '#call' do
     subject { described_class.new(notebook_url: notebook_url) }
-    it { expect(subject.call).not_to be_nil }
+
```
Comment (nemaniarjun) [inline spec/services/notebook_service_spec.rb]: @piotrekpasciak I think we can remove this test for now, crowdai gitlab is going to go down and this is not required anymore. 

@skbly7 @spMohanty What was this notebook service thing? seems to be used in articles. Is it needed in AIcrowd or should we :fire: it?

**Comment by nemaniarjun** [comment]
@piotrekpasciak this seems great. 
Pending decision on the Notebook service and if we need it or not, but other than that LGTM! 
:shipit: :rocket:

File: spec/services/notebook_service_spec.rb (hunk 1 above)
Comment (spMohanty) [reply spec/services/notebook_service_spec.rb]: @nemaniarjun @piotrekpasciak : The notebook service stuff can be burned ! Its not relevant at the moment.

Comment (piotrekpasciak) [reply spec/services/notebook_service_spec.rb]: @nemaniarjun  @spMohanty  I'll submit separate PR for removal of that service.

**Pull request merged**